src/pyrokinetics/local_geometry/local_geometry.py

```diff
@@ -528,25 +527,13 @@ def get_bunit_over_b0(self, ntheta: int = 256):
 
         """
 
-        theta = np.linspace(0, 2 * pi, ntheta)
+        theta = np.linspace(0, 2 * pi, ntheta, endpoint=False)
+        dtheta = np.diff(theta, append=theta[0] + 2 * np.pi)
 
         R, Z = self.get_flux_surface(theta=theta)
+        dRdtheta, dRdr, dZdtheta, dZdr = self.get_RZ_derivatives(theta)
 
-        # TODO Numpy roll doesn't work on pint=0.23 quantities
-        if isinstance(R, PyroQuantity):
-            l_units = R.units
-            R = R.m
-            Z = Z.m
-        else:
-            l_units = 1
-
-        dR = (np.roll(R, 1) - np.roll(R, -1)) / 2.0 * l_units
-        dZ = (np.roll(Z, 1) - np.roll(Z, -1)) / 2.0 * l_units
-
-        R *= l_units
-        Z *= l_units
-
-        dL = np.sqrt(dR**2 + dZ**2)
+        dL = np.sqrt(dRdtheta**2 + dZdtheta**2) * dtheta
 
         R_grad_r = R * self.get_grad_r(theta)
         integral = np.sum(dL / R_grad_r)
```

Comment:
I suppose if one really wanted to get the "correct" result we could use something like `quad` to integrate a `def dL(theta): dRdtheta, _, dZdtheta, _ = self.get_RZ_derivatives(theta) ; return np.sqrt(dRdtheta**2 + dZdtheta**2)` type function?

Reply:
This might at least allow one to measure the error and report if it looks like we're under-resolved?

Reply:
Quick gander at it using scipy gets you `3.548375666156662 dimensionless`. I guess this is better than specifying ntheta